### ISSUE TITLE
crypto/x509: verification fails with "cannot parse dnsName" in intermediate

### ISSUE THREAD

**Issue by trunet** [kubernetes/client-go#371]
Loading an intermediate certificate, is generating this error:
```
E0226 11:51:47.736103       7 config.go:330] Expected to load root CA config from /var/run/secrets/kubernetes.io/serviceaccount/ca.crt, but got err: error reading /var/run/secrets/kubernetes.io/serviceaccount/ca.crt: x509: cannot parse dnsName "Trunet.co AMS Kubernetes Intermediate CA"
```

My intermediate CA generated by hashicorp vault, contains:
```
X509v3 Basic Constraints: critical
    CA:TRUE
X509v3 Subject Alternative Name: 
    DNS:Trunet.co AMS Kubernetes Intermediate CA
```

This will probably be fixed on https://github.com/golang/go/issues/23995

**Comment by sttts** [comment]
So this is no client-go error, but Go?

**Comment by ericchiang** [comment]
"Trunet.co AMS Kubernetes Intermediate CA" doesn't look like a DNS name. Why would you want that value as a SANs DNS?

It sounds like this is getting fixed in 1.10.1. Since you can work around this today by making your DNS values look like actual DNS values, I don't think we should provide an external work around.

**Comment by trunet** [comment]
from: https://github.com/golang/go/issues/23995#issuecomment-368784560

> This will affect all CA certs that have been generated by HashiCorp Vault 0.9.3 or earlier. Those versions always put the CN also into the DNS alt names, without checking if it actually is a DNS name. hashicorp/vault#3953 fixed that in anticipation of the change in Go.
>
> Relaxing the verification for CA certs would greatly reduce pain for Vault operators because they wouldn't have to re-issue all of their cert chains.
